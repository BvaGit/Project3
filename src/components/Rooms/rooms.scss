@import "../../utils/variables.scss";

.rooms {
  padding: 0 $padding_sm;
  height: 350px;
  overflow-y: scroll;
  &__message {
    text-align: center;
  }
  &::-webkit-scrollbar {
    border-radius: 10px;
    width: 10px;
    height: 40px;
    background-color: $l-body-text-color;
  }

  &::-webkit-scrollbar-thumb {
    border-radius: 10px;
    background-color: $cross;
  }

  &::-webkit-scrollbar-track {
    -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.2);
    border-radius: 10px;
    background-color: $l-body-text-color;
  }
}

.theme-dark {
  .rooms {
    &__message {
      color: $l-body-text-color;
    }
  }
}
