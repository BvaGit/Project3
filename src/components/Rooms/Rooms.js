import React from 'react';
import SingleRoom from '../SingleRoom';
import './rooms.scss';


const Rooms = () => (
    <div className="rooms">
        <SingleRoom/>
        <SingleRoom/>
        <SingleRoom/>
    </div>
)

export default Rooms;