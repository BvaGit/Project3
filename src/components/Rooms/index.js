import Rooms from "./Rooms";
import { connect } from "react-redux";

import { getUserChats } from "../../store/rooms/asyncActions";
import { setActiveRoom } from "../../store/rooms/actions";
import { getUserID } from "../../store/user/selectors";
import { getRoomsArray } from "../../store/rooms/selectors";

const mapStateToProps = (state) => ({
  userId: getUserID(state),
  rooms: getRoomsArray(state),
});

const mapDispatchToProps = (dispatch) => ({
  getUserChats: (userId) => dispatch(getUserChats(userId)),
  setActiveRoom: (chatId) => dispatch(setActiveRoom(chatId)),
});

export default connect(mapStateToProps, mapDispatchToProps)(Rooms);
