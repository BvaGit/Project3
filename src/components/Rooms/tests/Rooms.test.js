import React from 'react';
import { shallow, mount } from 'enzyme';
import Rooms from '../Rooms';
import { Provider } from 'react-redux';
import { IntlProvider } from "react-intl";




describe('Rooms', () => {
    let props;
    it("Should match snapshot", () => {
        props={rooms:[]}
        const component = shallowBiggerSmart(<Rooms {...props}/>)
        expect(component.html()).toMatchSnapshot()
    })
})

export const shallowBiggerSmart = (component, store) => {
    const core = store
        ? <Provider store={store}>{component}</Provider>
        : component;
    return shallow(<IntlProvider >core</IntlProvider>);
};