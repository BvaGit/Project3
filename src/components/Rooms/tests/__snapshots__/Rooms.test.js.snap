// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rooms Should match snapshot 1`] = `"core"`;
